utils/sinv.f.in

```diff
@@ -0,0 +1,106 @@
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+      PROGRAM SINV
+ 
+      PARAMETER (MAXA=16000000)
+      PARAMETER (MAXS=1000)
+ 
+      CHARACTER(255) FILE   
+      CHARACTER(100) cmtdir 
+      CHARACTER(8)  SUBSET
+      CHARACTER     ci*16,cj*80
+      DIMENSION     isat(0:maxs,0:maxs)  
+      real(8)       arr(2,maxa),said(maxa),siid(maxa)
+ 
+      DATA BMISS  /10E10/
+      DATA LUNBF  /20/
+ 
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+ 
+      isat=0  
+      jsat=0  
+      said=0
+      ssid=0
+      close(6); open(6,recl=130)
+
+!  get filename argument
+
+      NARG=IARGC()
+      IF(NARG/=1) THEN
+        write(6,*)'Usage: sinv <satbufrfile> will print inventory of satellites by platform and instrument'
+        CALL EXIT(2)
+      ENDIF
+      call getarg(1,file)
+      file = TRIM(adjustl(file))
+
+      call mtinfo("@MASTER_TABLE_DIR@",3,4)
+
+      open(lunbf,file=file,form='unformatted')
+      CALL OPENBF(LUNBF,'IN',LUNBF)
+      call codflg('Y')
+
+      call ufbtab(lunbf,said,1,maxa,nret,'SAID')
+      !print*,nret
+      call ufbtab(lunbf,siid,1,maxa,nrex,'SIID')
+      !print*,nrex
+
+      call readmg(lunbf,subset,idate,iret)
+
+      do n=1,max(nret,nrex)
+      i = said(n)
+      j = siid(n)  
+      if(i>maxs.or.i<0) i=0                 
+      if(j>maxs.or.j<0) j=0                 
+      isat(i,j) = isat(i,j)+1
+      enddo
+
+      !print1,'satellite     ','instrument    ','   count'
+      !print'(40("-"))'
+      print*
+1     format(a14,2x,a14,2x,a8)
+      do i=0,1000
+      do j=0,1000
+      if(isat(i,j).gt.0) then 
+         jsat=jsat+isat(i,j)
+         call satcode(lunbf,i,ci,j,cj)
+         write(6,'(i3.3,2x,a16,2x,i10,2x,i3.3,a80)')i,ci,isat(i,j),j,trim(cj)
+      endif
+      enddo
+      enddo
+
+      print*; print'(23x,i10)',jsat; print*
+
+      stop
+      end
+!-----------------------------------------------------------------------
+!  looks up BUFR code table values for SAID (said) and SIID (instrument)
+!-----------------------------------------------------------------------
+      subroutine satcode(lunit,icode,csad,jcode,csid)
+
+      character(*) :: csad,csid
+
+      csad=' '; csid=' '
+
+!  call routines to look  up the said and siid 
+
+      if(icode>0) call getcfmng(lunit,'SAID',icode,'  ',-1,csad,len,iret)
+      if(jcode>0) call getcfmng(lunit,'SIID',jcode,'  ',-1,csid,len,iret)
+
+      return
+      end
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+      SUBROUTINE BORT(STR)
```

Comment:
maybe just add a comment here that this version of bort is superseding the library version of the subroutine of the same name

Reply:
Its for debugging, I could take it out.